mobile/src/components/common/input/input.tsx

```diff
@@ -27,24 +28,42 @@ const Input = <T extends FormControlValues>({
   control,
   errors,
   placeholder,
+  isSecure,
 }: Props<T>): ReactElement => {
   const { field } = useFormControl({ name, control });
+  const [visibility, setVisibility] = useState(isSecure);
 
   const { value, onChange, onBlur } = field;
   const error = errors[name]?.message as string;
 
+  const handleOnPress = (): void => {
+    setVisibility(!visibility);
+  };
+
   return (
     <View>
       <Text style={styles.label}>{label}</Text>
-      <TextInput
-        value={value}
-        placeholder={placeholder}
-        placeholderTextColor={AppColor.TEXT.GRAY_200}
-        onChangeText={onChange}
-        onBlur={onBlur}
-        style={styles.input}
-        secureTextEntry={name === 'password'}
-      />
+      <View>
+        <TextInput
+          value={value}
+          placeholder={placeholder}
+          placeholderTextColor={AppColor.TEXT.GRAY_200}
+          onChangeText={onChange}
+          onBlur={onBlur}
+          style={styles.input}
+          secureTextEntry={visibility}
+        />
+        {isSecure && (
```

Comment:
Create separate component InputSecure that will wrap usual input

Reply:
#### If divided into different components in this way:
  - need to duplicate props in each input
  - still need to transfer isSecure to a regular input
  - there is no way to correctly position the button

<img src="https://i.gyazo.com/8c0f468278efcec7399c5e34aafb2c42.png" width="100%">

#### If needed, we can create a separate component from scratch without wrapping the existing input.

Reply:
It should be a separate component, because it has additional logic that we don't need in usual inputs, you can add iconRight property to input, to have an ability to position it

Reply:
Sorry, I can't figure out what it should look like :face_with_head_bandage:
If I understand correctly you want the button with the icon to be inside the Input component then we will have to pass the onPress function and the IsSecure props to the Input component.
All that we take out to another component is useState

Reply:
Let's do it in simplest way for now, add isSecure property to Input , and reuse Input in InputSecur. To align icon in it just set fixed top value